app/PairNumber.js

```diff
@@ -0,0 +1,34 @@
+import {MajorColorNames} from './MajorColorNames.js';
+import {MinorColorNames} from './MinorColorNames.js';
+
+export class PairNumber {
+    constructor(pair) {
+        this.pair = pair;
+    }
+
+    //This function returns the pair number for the given color pair.
+    getPairNumberFromColor(pair) {
+        let majorIndex = -1;
+        for (let i = 0; i < MajorColorNames.length; i++)
+            {
+                if (MajorColorNames[i] == pair.majorColor) {
+                    majorIndex = i;
+                    break;
+                }
+            }
+
+        let minorIndex = -1;

```

Comment:
Notice the repeated pattern in the code: initialize to -1, iterate and compare. Only the minor and major are different. Try moving this into a function and call it in both contexts to remove the duplication.